src/test.js

```diff
@@ -623,25 +604,61 @@ function skip( testName ) {
 }
 
 // Will be exposed as QUnit.only
-function only( testName, expected, callback, async ) {
+function only( testName, callback ) {
 	var newTest;
 
 	if ( focused )  { return; }
 
 	QUnit.config.queue.length = 0;
 	focused = true;
 
-	if ( arguments.length === 2 ) {
-		callback = expected;
-		expected = null;
-	}
-
 	newTest = new Test( {
 		testName: testName,
-		expected: expected,
-		async: async,
 		callback: callback
 	} );
 
 	newTest.queue();
 }
+
+function internalStop() {
+
+	// If a test is running, adjust its semaphore
+	config.current.semaphore += 1;
+
+	pauseProcessing();
+}
+
+function internalStart() {
+
+	// If a test is running, adjust its semaphore
+	config.current.semaphore -= 1;
+
+	// If semaphore is non-numeric, throw error
+	if ( isNaN( config.current.semaphore ) ) {
```

Comment:
21c254a I tried this refactoring, it's working and it avoids using the config.current using test directly whenever is possible. I'll do a inline comment to show the point I'm not satisfied